src/EventGrid/EventGrid/Domain/NewAzureEventGridDomain.cs

```diff
@@ -120,6 +120,62 @@ public class NewAzureEventGridDomain : AzureEventGridCmdletBase
         [ValidateNotNullOrEmpty]
         public string PublicNetworkAccess { get; set; } = EventGridConstants.Enabled;
 
+        /// <summary>
+        /// DisableLocalAuth
+        /// </summary>
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = EventGridConstants.DisableLocalAuthHelp,
+            ParameterSetName = DomainNameParameterSet)]
+        [ValidateSet(EventGridConstants.Enabled, EventGridConstants.Disabled, IgnoreCase = true)]
```

Comment:
Please remove ValidateSet. For switch parameter DisableLocalAuth, -DisableLocalAuth means disable, otherwise it means enable.

Reply:
fixed